scripts/linters/pre_commit_linter.py

```diff
@@ -459,37 +471,40 @@ def _get_all_filepaths(input_path, input_filenames, input_shard):
     elif input_shard:
         if input_shard != OTHER_SHARD_NAME:
             all_filepaths = _get_filepaths_from_non_other_shard(
-                input_shard)
+                input_shard, name_space=name_space)
         else:
-            all_filepaths = _get_filepaths_from_other_shard()
+            all_filepaths = _get_filepaths_from_other_shard(
+                name_space=name_space)
     else:
         all_filepaths = _get_changed_filepaths()
-    all_filepaths = [
+    all_matching_filepaths = [
         filename for filename in all_filepaths if not
         any(
-            fnmatch.fnmatch(filename, pattern) for pattern
+            fnmatch.fnmatch(filename, pattern) for pattern  # pylint: disable=comprehension-escape
```

Comment:
Why the pylint pragma?

Reply:
This is a false positive. It is reporting that some variable (`filename` or `pattern`) is escaping the comprehension, which is not true here since we are still inside the comprehension.

Reply:
How odd. Can you raise this with the dev workflow team and also get a bit more of a handle on the issue so that you can write a TODO comment? E.g. change any(...) to just "filename == 'abc'" and see if the error still persists, then change i to "any(fnmatch.fnmatch(filename, 'xyz'))" and if there's no error then it's probably pattern. Does it arise when you don't break after `any(`?

Reply:
Done, it seems to be caused by the `pattern`, added a TODOD comment.

cc @DubeySandeep

Reply:
Looks like this bug is reported here https://github.com/PyCQA/pylint/issues/3487 and it's reproduciable only in py3

(I'll update our issue thread with the same!)